corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js

```diff
@@ -387,6 +387,11 @@ hqDefine("cloudcare/js/form_entry/entrycontrols_full", function () {
         EntryArrayAnswer.call(this, question, options);
         self.templateType = 'select';
         self.choices = question.choices;
+        self.choices_v2 = _.map(question.choices_v2(), function (choice) {
```

Comment:
Do `MultiSelectEntry` and/or `ChoiceLabelEntry` need this logic?

Alternatively, is it possible to mutate `choices` in `Container.prototype.fromJS` and avoid having a `choices_v2` in the entry classes?

Reply:
So `MultiSelectEntry` is also updated to use `choices_v2`, and the `ChoiceLabelEntry` is able to use the question level `caption_image` field. 

As far as mutating `choices`, I read about [knockout mappings](https://knockoutjs.com/documentation/plugins-mapping.html) to understand that a bit better, and tried to override the create option for `choices` while ignoring `choices_v2`, but `options.parent` is an empty array so I'm unable to determine what the question datatype is which would be necessary to choose `choices` or `choices_v2`. Am I missing anything there?

Reply:
No suggestions, I don't know knockout mapping options all that well. Since you explained `choices_v2` only gets defined in 2 places and doesn't need to be defined in the additional 2 I was thinking it did, this seems fine.